cmd/kubernetesDeploy.go

```diff
@@ -51,21 +51,44 @@ func runHelmDeploy(config kubernetesDeployOptions, utils kubernetes.DeployUtils,
 	if err != nil {
 		log.Entry().WithError(err).Fatalf("Container registry url '%v' incorrect", config.ContainerRegistryURL)
 	}
-	//support either image or containerImageName and containerImageTag
-	containerImageName := ""
-	containerImageTag := ""
 
-	if len(config.Image) > 0 {
-		containerImageName, containerImageTag, err = splitFullImageName(config.Image)
-		if err != nil {
-			log.Entry().WithError(err).Fatalf("Container image '%v' incorrect", config.Image)
+	helmValues := helmValues{}
+
+	if len(config.ImageNames) > 0 {
+		if len(config.ImageNames) != len(config.ImageNameTags) {
+			return fmt.Errorf("number of imageNames and imageNameTags must be equal")
+		}
+		for i, key := range config.ImageNames {
+			name, tag, err := splitFullImageName(config.ImageNameTags[i])
+			if err != nil {
+				log.Entry().WithError(err).Fatalf("Container image '%v' incorrect", config.ImageNameTags[i])
+			}
+
+			helmValues.add(joinKey("image", key, "repository"), fmt.Sprintf("%v/%v", containerRegistry, name))
+			helmValues.add(joinKey("image", key, "tag"), tag)
 		}
-	} else if len(config.ContainerImageName) > 0 && len(config.ContainerImageTag) > 0 {
-		containerImageName = config.ContainerImageName
-		containerImageTag = config.ContainerImageTag
 	} else {
-		return fmt.Errorf("image information not given - please either set image or containerImageName and containerImageTag")
+		//support either image or containerImageName and containerImageTag
+		containerImageName := ""
+		containerImageTag := ""
+		if len(config.Image) > 0 {
+			containerImageName, containerImageTag, err = splitFullImageName(config.Image)
+			if err != nil {
+				log.Entry().WithError(err).Fatalf("Container image '%v' incorrect", config.Image)
+			}
+		} else if len(config.ContainerImageName) > 0 && len(config.ContainerImageTag) > 0 {
+			containerImageName = config.ContainerImageName
+			containerImageTag = config.ContainerImageTag
+		} else {
+			return fmt.Errorf("image information not given - please either set image or containerImageName and containerImageTag")
+		}
+		helmValues.add("image.repository", fmt.Sprintf("%v/%v", containerRegistry, containerImageName))
+		helmValues.add("image.tag", containerImageTag)
+
+		helmValues.add(joinKey("image", containerImageName, "repository"), fmt.Sprintf("%v/%v", containerRegistry, containerImageName))
+		helmValues.add(joinKey("image", containerImageName, "tag"), containerImageTag)
```

Comment:
Do we need this `else` block at all, given that `if len(config.ImageNames) > 0` should always be true for recent versions of both `kanikoExecute` and `cnbBuild`.
They do both publish single and multiple images via `container/imageNameTags`, don't they?

cc @OliverNocon